collectors/ebpf_process.plugin/ebpf_process.c

```diff
@@ -853,11 +867,185 @@ static int load_collector_file(char *path) {
+            "\n"
+            " --global or -g    disable charts per application.\n"
+            "\n"
+            " --all or -a       Enable all chart groups (global and apps), unless -g is also given.\n"
+            "\n"
+            " --net or -n       Enable network viewer charts.\n"
+            "\n"
+            " --process or -p   Enable charts related to process run time.\n"
+            "\n"
+            " --return or -r    Run the collector in return mode.\n"
+            "\n"
+            , VERSION
+            , (year >= 116)?year + 1900: 2020
+    );
+}
+
+static void parse_args(int argc, char **argv)
 {
-    (void)argc;
-    (void)argv;
+    int enabled = 0;
+    int disable_apps = 0;
+    int freq = 0;
+    int c;
+    int option_index = 0;
+    static struct option long_options[] = {
+        {"help",     no_argument,    0,  'h' },
+        {"version",  no_argument,    0,  'v' },
+        {"global",   no_argument,    0,  'g' },
+        {"all",      no_argument,    0,  'a' },
+        {"net",      no_argument,    0,  'n' },
+        {"process",  no_argument,    0,  'p' },
+        {"return",   no_argument,    0,  'r' },
+        {0, 0, 0, 0}
+    };
+
+    if (argc > 1) {
+        int n = (int)str2l(argv[1]);
+        if(n > 0) {
+            freq = n;
+        }
+    }
+
+    while (1) {
+        c = getopt_long(argc, argv, "hvganpr",long_options, &option_index);
+        if (c == -1)
+            break;
+
+        switch (c) {
+            case 'h': {
+                ebpf_print_help();
+                exit(0);
+            }
+            case 'v': {
+                printf("ebpf.plugin %s\n", VERSION);
+                exit(0);
+            }
+            case 'g': {
+                disable_apps = 1;
+                ebpf_disable_apps();
+#ifdef NETDATA_INTERNAL_CHECKS
+                info("EBPF running with global chart group, because it was started with the option \"--global\" or \"-g\".");
+#endif
+                break;
+            }
+            case 'a': {
+                ebpf_enable_all_charts(disable_apps);
+#ifdef NETDATA_INTERNAL_CHECKS
+                info("EBPF running with all chart groups, because it was started with the option \"--all\" or \"-a\".");
+#endif
+                break;
+            }
+            case 'n': {
+                enabled = 1;
+                ebpf_enable_chart(1, disable_apps);
+#ifdef NETDATA_INTERNAL_CHECKS
+                info("EBPF enabling \"NET\" charts, because it was started with the option \"--net\" or \"-n\".");
+#endif
+                break;
+            }
+            case 'p': {
+                enabled = 1;
+                ebpf_enable_chart(0, disable_apps);
+#ifdef NETDATA_INTERNAL_CHECKS
+                info("EBPF enabling \"PROCESS\" charts, because it was started with the option \"--process\" or \"-p\".");
+#endif
+                break;
+            }
+            case 'r': {
+                mode = 0;
```

Comment:
can we use enum here or at least define? For somebody who not have an overview of the code, it might not be obvious what mode = X means *(well it is clear here as there is info message with `return mode`)*.

Reply:
No need to change it in this PR but for future i think it is something to improve

Reply:
The last commit brought all changes you required @underhood. Thank you for the suggestions!